src/fs.c

```diff
@@ -0,0 +1,303 @@
+    while (iterator) {
+        console_printf("%s -> ", ((struct fs_allowance *)iterator)->path);
+        iterator = iterator->next;
+    }
+    console_printf("\n");
+}
+
+void fs_sys_init_open_files_table() {
+    int i;
+    for (i = 0; i < MAX_OS_OPEN_FILES; i++) {
+        open_files_table[i].filep = 0;  // Set each file pointer to null
+    }
+}
+
+void fs_init_security(struct process *p) {
+    // Initialize all file descriptor slots to 0
+    int i;
+    for (i = 0; i < PROCESS_MAX_OPEN_FILES; i++) {
+        p->fd_table[i].is_open = 0;
+        p->fd_table[i].ptr = 0;
+    }
+
+    //Allowances alloc'ing.
+    struct fs_allowance *allowance = kmalloc(sizeof(*allowance));
+    strcpy(allowance->path, "/");
+    allowance->do_allow_below = 1;
+
+    //Set the lone permission in the list to root (that is all of the fs)
+    list_push_head(&(p->fs_allowances_list), (struct list_node *)allowance);
+}
+
+// Look at the OS's table of open files to see if read write conflicts
+// might be occurring amongst many processes if you allow it to be opened
+bool fs_check_open_conflict(const char *path, const char *mode) {
+    return 1;
+}
+
+int map_media_to_driver_id(int media) {
+    return ISO;
+}
+
+int mode_str_to_int(const char *mode) {
+    if (strcmp(mode, "a") == 0) {
+        return APPEND;
+    }
+    if (strcmp(mode, "w") == 0) {
+        return WRITE;
+    }
+    if (strcmp(mode, "r") == 0) {
+        return READ;
+    }
+    if (strcmp(mode, "ar") == 0) {
+        return READ | APPEND;
+    }
+    if (strcmp(mode, "ra") == 0) {
+        return READ | APPEND;
+    }
+    if (strcmp(mode, "rw") == 0) {
+        return READ | WRITE;
+    }
+    if (strcmp(mode, "wr") == 0) {
+        return READ | WRITE;
+    }
+    return 0;
+}
+
+struct fs_agnostic_file *create_fs_agnostic_file(enum ata_kind ata_type, uint8_t ata_unit, const char *path, uint8_t mode) {
+    int next_open_slot;
+    for (next_open_slot = 0; next_open_slot < MAX_OS_OPEN_FILES; next_open_slot++) {
+        if (!open_files_table[next_open_slot].filep) {  // If the file pointer hasn't been assigned, this is an unused file slot
+            break;
+        }
+    }
+    struct fs_agnostic_file *new_file = &(open_files_table[next_open_slot]);
+    new_file->ata_unit = ata_unit;
+    new_file->ata_type = ata_type;
+    new_file->mode = mode;
+    strcpy(new_file->path, path);
+    switch (ata_type) {
+        case ISO:
+            new_file->filep = (void *)iso_fopen(path, ata_unit);
+            break;
+        default:
+            kfree(new_file);
+            return 0;
+    }
+    if (!new_file->filep) {
+        //media failed to open the file and returned 0
+        kfree(new_file);
+        return 0;
+    }
+    return new_file;
+}
+
+int32_t fs_open(const char *path, const char *mode) {
+    //checks both owner permissions (dummy true) and process permissions
+    uint32_t success = fs_security_check(path);
+    if (success < 1) {
```

Comment:
As `fs_security_check` should return a signed value, not sure if C would convert return values automatically, but for correctness sake `success` should be signed.